beacon-chain/forkchoice/doubly-linked-tree/types.go

```diff
@@ -38,6 +38,7 @@ type Store struct {
 	checkpointsLock               sync.RWMutex
 	genesisTime                   uint64
 	highestReceivedSlot           types.Slot                            // The highest received slot in the chain.
+	highestReceivedRoot           [fieldparams.RootLength]byte          // The highest received root in the chain.
```

Comment:
Instead of adding these 40 bytes and updating this root every inserted block, you better track `highestReceivedNode` with a pointer to the corresponding node, it's only 8 bytes and you do not need to reallocate the root

Reply:
That makes sense, thanks!